src/main.rs

```diff
@@ -305,68 +300,78 @@ async fn handle_web_req(
 
             let mut lnd = handler.client;
 
-            let headers = Headers(req.headers().to_owned());
-            let query = {
-                let uri = req.uri();
-                if let Some(query) = uri.query() {
-                    Some(&query.to_owned());
-                }
-                None
-            };
-            let body = req.into_body();
-            let bytes = hyper::body::to_bytes(body).await?;
-            dbg!(&bytes); // this is correct by my accounts
-            let reader = &*bytes;
-            let proposal = UncheckedProposal::from_request(reader, query, headers).unwrap();
+            // extract original PSBT from HTTP request
+            // TODO: `UncheckedProposal` is not really the proposal, but the original PSBT? Is the naming wrong?
+            // TODO: Unneeded and/or undocumented traits in bip-78 library? (`Proposal`, `Headers`)
```

Comment:
Proposal is probably unecessary, but I imagine as a way to extract `psbt() -> Psbt` between check states if necessary. That might only be necessary at the fully checked / unlockedProposal stage. We can play with the interface to arrive at v1.0

Reply:
This is also an issue for the bip78.